[doc] Update dev_install.md to reflect build.py usage (#7848)
taichi-dev/taichi: docs/lang/articles/contribution/dev_install.md

:::

:::note

With the release of Taichi v1.6.0, a comprehensive build environment preparation script (aka. `build.py` or `ti-build`) has been introduced. This script significantly simplifies the process of configuring a suitable build or development environment.

This guide will focus on the `build.py` approach. If you prefer to use the conventional method, you can refer to the previous Developer Installation document.

:::

## Prerequisites

````mdx-code-block
<Tabs
  defaultValue="linux"
  values={[
    {label: 'Linux', value: 'linux'},
    {label: 'Mac', value: 'mac'},
    {label: 'Windows', value: 'windows'}
  ]}>

<TabItem value="linux">

| Category                      | Prerequisites                                                        |
| :---------------------------- | :------------------------------------------------------------------- |
| Linux distribution            | Anything recent enough, e.g. Ubuntu 20.04                            |

> | Python                        | 3.6+, with a usable distutils(`python3-distutils` package on Ubuntu) |

Reviewer: Do we support >python3.10?

Author: Yes, `3.6+` means 3.6 or later

</TabItem>

<TabItem value="mac">

| Category                       | Prerequisites                            |
| :----------------------------- | ---------------------------------------- |
| macOS                          | macOS Big Sur or later                   |
| Python                         | 3.6+ (should be readily available)       |
| Command line tools for Xcode   | Run `xcode-select --install` to install  |

</TabItem>

<TabItem value="windows">

| Category        | Prerequisites                                                                   |
| :-------------- | ------------------------------------------------------------------------------- |
| Windows         | Windows 7/8/10/11                                                               |
| Python          | 3.6+                                                                            |
| Visual Studio   | Visual Studio 2022 (any edition) with "Desktop Development with C++" component. |

</TabItem>

</Tabs>
````

### Install Compiler

<blockquote>

Taichi supports building from source with Clang++ >= 10.0 and MSVC from VS2022.

> For macOS developers, it is recommended to use AppleClang, which comes with the Command Line Tools for Xcode. You can install them by running `xcode-select --install`. Alternatively, a complete installation of Xcode.app is also an option.

Reviewer: ```suggestion
For macOS developers, it is recommended to use AppleClang, which comes with the Command Line Tools for Xcode. You can install them by running `xcode-select --install`. Alternatively, you can also install Xcode.app from the Apple Store.
```

<TabItem value="linux">

  ```shell
  git clone --recursive https://github.com/taichi-dev/taichi

  cd taichi

  # Customize with your own needs
  export TAICHI_CMAKE_ARGS="-DTI_WITH_VULKAN:BOOL=ON -DTI_WITH_CUDA:BOOL=ON"

  # Uncomment if you want to use a different compiler
  # export CC=/path/to/clang
  # export CXX=/path/to/clang++

  # export DEBUG=1 # Uncomment it if you wish to keep debug information.

  # This would build a wheel file in `dist/` folder
  ./build.py

  # Alternatively, this would drop into a shell with complete build environment,
  ./build.py --shell
  # and then you could install Taichi in development mode

>   python3 setup.py develop

Reviewer: I'd suggest put the alternative method in a separate section/code-block, so it's not gonna confuse the beginners.
